File: backend/app/schemas/order.py
"""订单Pydantic Schema"""

from typing import List, Optional

from pydantic import BaseModel, Field


class OrderCreate(BaseModel):
    """订单创建Schema"""

    symbol: str = Field(..., description="标的代码")
    market: str = Field("crypto", description="市场类型：crypto / a_share")
    side: str = Field(..., description="买卖方向：BUY / SELL")
    type: str = Field("MARKET", description="订单类型：MARKET / LIMIT")
    price: float = Field(0.0, description="下单价格(LIMIT单必填)")
    quantity: float = Field(..., gt=0, description="下单数量")


class OrderResponse(BaseModel):
    """订单响应Schema"""

    id: str = Field(..., description="记录ID")
    order_id: str = Field(..., description="订单唯一ID")
    symbol: str = Field(..., description="标的代码")
    market: str = Field(..., description="市场类型")
    side: str = Field(..., description="买卖方向")
    type: str = Field(..., description="订单类型")
    price: float = Field(..., description="下单价格")
    quantity: float = Field(..., description="下单数量")
    status: str = Field(..., description="订单状态")
    created_at: str = Field("", description="创建时间")
    updated_at: str = Field("", description="更新时间")

    model_config = {"from_attributes": True}


class OrderListResponse(BaseModel):
    """订单列表响应Schema"""

    total: int = Field(0, description="总数")
    items: List[OrderResponse] = Field(default_factory=list, description="订单列表")

File: frontend/src/api/order.ts
/** 订单API调用 - createOrder, getOrders, cancelOrder */

import apiClient from "./client";
import type { ApiResponse, Order } from "@/types";

/** 订单创建参数 */
interface OrderCreateParams {
  symbol: string;
  market?: string;
  side: "BUY" | "SELL";
  type?: "MARKET" | "LIMIT";
  price?: number;
  quantity: number;
}

/** 订单列表响应 */
interface OrderListData {
  total: number;
  items: Order[];
}

/**
 * 创建订单（手动下单）
 * @param data 订单参数
 * @returns 创建的订单
 */
export async function createOrder(
  data: OrderCreateParams,
): Promise<ApiResponse<Order>> {
  const response = await apiClient.post<ApiResponse<Order>>("/orders", data);
  return response.data;
}

/**
 * 获取订单列表（分页）
 * @param params 查询参数
 * @returns 订单列表响应
 */
export async function getOrders(params?: {
  symbol?: string;
  status?: string;
  page?: number;
  page_size?: number;
}): Promise<ApiResponse<OrderListData>> {
  const response = await apiClient.get<ApiResponse<OrderListData>>("/orders", {
    params,
  });
  return response.data;
}

/**
 * 获取订单详情
 * @param orderId 订单ID
 * @returns 订单详情
 */
export async function getOrder(
  orderId: string,
): Promise<ApiResponse<Order>> {
  const response = await apiClient.get<ApiResponse<Order>>(
    `/orders/${orderId}`,
  );
  return response.data;
}

/**
 * 取消订单
 * @param orderId 订单ID
 * @returns 取消后的订单状态
 */
export async function cancelOrder(
  orderId: string,
): Promise<ApiResponse<Order>> {
  const response = await apiClient.delete<ApiResponse<Order>>(
    `/orders/${orderId}`,
  );
  return response.data;
}
